src/app/workbench/WorkbenchViews.tsx

```diff
@@ -22,127 +41,66 @@ export function WorkbenchViews({ index, type }: IWorkbenchViewsProps) {
   const { views, selection, commentsOpen, itemIDType } = ordino.workbenches[index];
   const [setRef] = useCommentPanel({ selection, itemIDType, commentsOpen, isFocused: type === EWorkbenchType.FOCUS });
 
-  let wb = null;
+  const [mosaicState, setMosaicState] = useState<any>(null);
+  const [mosaicViewCount, setMosaicViewCount] = useState<number>(1);
+
+  const [mosaicDrag, setMosaicDrag] = useState<boolean>(false);
+
+  React.useEffect(() => {
+    if (mosaicState === null) {
+      setMosaicState(views[0].uniqueId);
```

Comment:
Why don't you add the view directly when initializing the state in line 44? Then you could remove the if condition here.

```
const [mosaicState, setMosaicState] = useState<any>(views[0].uniqueId);
```

Reply:
ah, ya, this code was an artifact of doing it a worse way. Agreed! Done